src/app/components/forms/game/video/video.ts

```diff
@@ -36,17 +58,9 @@ export default class FormGameVideo extends BaseForm<FormModel> implements FormOn
 	onUrlChange(url: string) {
 		// Check if we need to scrub out anything from the URL.
 		// Will be the case if they entered in a full URL such as http://www.youtube.com/watch?v=something, etc.
-		if (url) {
-			const youtubeMatch = url.match(GameVideo.REGEX.YOUTUBE);
-			const vimeoMatch = url.match(GameVideo.REGEX.VIMEO);
-
-			if (youtubeMatch) {
-				this.setField('type', GameVideo.TYPE_YOUTUBE);
-				this.setField('url', youtubeMatch[4]);
-			} else if (vimeoMatch) {
-				this.setField('type', GameVideo.TYPE_VIMEO);
-				this.setField('url', vimeoMatch[4]);
-			}
+		if (url && this.hasValidVideoUrl) {
+			this.setField('type', this.videoData!.type);
+			this.setField('url', this.videoData!.id);
```

Comment:
getters are only cached for vue rendering. the videoData would still be called 3 times in this. lets just get the videoData once and work off of it, yeah?